node/service/src/parachains_db/upgrade.rs

```diff
@@ -55,10 +55,31 @@ impl From<Error> for io::Error {
 	}
 }
 
-/// Try upgrading parachain's database to the current version.
-pub(crate) fn try_upgrade_db(db_path: &Path, db_kind: DatabaseKind) -> Result<(), Error> {
+/// Try upgrading parachain's database to a target version.
+pub(crate) fn try_upgrade_db(
+	db_path: &Path,
+	db_kind: DatabaseKind,
+	target_version: Version,
+) -> Result<(), Error> {
+	// Loop upgrades until we reach the target version
+	loop {
+		let version = try_upgrade_db_to_next_version(db_path, db_kind)?;
+		if version == target_version {
```

Comment:
might make sense to add some safety here so we don't loop forever because of a bug.